### PR TITLE
Fix python

### PR DESCRIPTION
**Pull request by jakirkham**
Related PR ( https://github.com/conda-forge/nipype-feedstock/pull/1 ). Fixes Python dependency and re-renders. Bumps build number.

### CODE REVIEW

**Comment by jakirkham** [comment]
Weird. I need to investigate something first.

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,9 @@
 language: objective-c
 
 env:
+  matrix:
+    
+    - CONDA_PY=35
```
Comment (jakirkham) [inline .travis.yml]: Seems to only include Python 3.5. @pelson, any idea why this is happening?

File: ci_support/run_docker_build.sh [hunk 2]
```diff
@@ -42,6 +42,9 @@ conda install --yes --quiet conda-forge-build-setup
 source run_conda_forge_build_setup
 
 # Embarking on 1 case(s).
+    set -x
+    export CONDA_PY=35
```
Comment (jakirkham) [inline ci_support/run_docker_build.sh]: Also only including Python 3.5 here.

**Comment by jakirkham** [comment]
Note that `appveyor.yml` was not updated by the re-rendering either.

**Comment by jakirkham** [comment]
This problem seems to go away when using `conda-build-all` version `0.12.0`.

**Comment by chrisfilo** [comment]
Users report Python being updated to version 3 when nipype package is
installed in a virtual environment set up for Python 2. Could this be
related?

**Comment by jakirkham** [comment]
So you were doing the right thing by adding `python` to the recipe. However, we have to update the CI scripts, which is what I was doing here. We also need to bump the build number.

In the process of updating the CI scripts, I discovered what appears to be a bug in the tools used to do that update. So I wanted to make some notes on it. That way @pelson and I can explore them later.

**Comment by jakirkham** [comment]
Alright, I have archived the broken packages so that no one should be able to install them and restarted all the CIs. This should pass.

**Comment by jakirkham** [comment]
So, I think we need to do this in a few other places.

**Comment by jakirkham** [comment]
I think we need to merge this PR ( https://github.com/conda-forge/prov-feedstock/pull/1 ) before we can fix these issues. Let's make sure that one tests out ok first.

**Comment by jakirkham** [comment]
Went ahead and restarted the CI here now that the `prov` builds are correct.

**Comment by jakirkham** [comment]
> Users report Python being updated to version 3 when nipype package is installed in a virtual environment set up for Python 2. Could this be related?

Basically if one does not specify what version of `python` one wants to build against, `conda-build` assumes it should be the same version of `python` as it itself is using. As we install Miniconda3, that ends up being Python 3.5. So, when you had added `python` it just built against the Python 3.5 version. Hence the upgrade phenomenon.

So we need a way to solve this issue. We do that by specifying the version to build against using the `CONDA_PY` environment variable. This ensures we build against the version we want. However, we have to insert this info into the CI scripts. The [`conda-smithy`]( https://github.com/conda-forge/conda-smithy ) tool does this with the help of [`conda-build-all`]( https://github.com/SciTools/conda-build-all/ ) by recognizing `python` is in the build requirements and specifying some supported versions to build against. Currently that is Python 2.7, 3.4, and 3.5. We do similar things with NumPy if you need to link to that (e.g. using Cython). We do this by running `conda smithy rerender` in the top-level directory of the repo and commit all changes it generates.

**Comment by jakirkham** [comment]
So this should get a proper `nipype` build out to your users. As mentioned I archived the bad builds so that they shouldn't pick them up. Please let me know if you have more issues.

**Pull request merged**